### ISSUE TITLE
Patch 1.26 isn't working on W10 1709 

### ISSUE THREAD

**Issue by azizzaamir** [snowie2000/mactype#411]
Tried the patch 1.26, also tried using CTP file still had no luck. Some apps aren't working but as soon as I stop the mactype, it works normally. Please reply.

**Comment by extratype** [comment]
Follow the following combination to use the latest MacTypePatch.
MacTypeInstaller_2016_0904_0.exe
MacType_CTP_170307_W10.zip
MacTypePatch_1.26.zip

**Comment by azizzaamir** [comment]
Shall I use old version of mactype installer as you said?

**Comment by extratype** [comment]
From the changelog and release dates, I guess that version is the latest compatible version. I had been using them, but I have switched to my own build.

**Comment by azizzaamir** [comment]
OK will try this, can you provide a link to your own build?

**Comment by extratype** [comment]
I am working on it, I'll upload it [here](https://github.com/extratype/mactype/releases) some days later. I tweaked 2017.628.0 a bit to meet my own taste.

**Comment by azizzaamir** [comment]
One more thing, how to apply CTP patch and patch 1.26 both? I'm afraid,I might be doing it wrong.

**Comment by extratype** [comment]
Apply CTP and then MacTypePatch.

**Comment by azizzaamir** [comment]
Sorry to say sir, but problem still persists. Tried everything

**Comment by extratype** [comment]
MacType conflicts with Avast. If that's the case, I recommend to choose only one of them.

**Comment by azizzaamir** [comment]
When I  set Userparams.ini in mactype wizard,everything works. But after a reboot nothing works.

**Comment by extratype** [comment]
You don't set UserParams.ini. You put it in MacType directory and it's automatically applied. It's separate settings from profiles in ini directory.

**Comment by azizzaamir** [comment]
Yes,tried doing that also didn't work, is this because I'm using avast security? Also please next time try to release a build fully compatible in W10 1709 with no patch needed

**Comment by extratype** [comment]
MacType is compatible with W10 1709, but not with Avast. MacTypePatch does not solve that issue.
Set HookChildProcesses=0 in your profile to make it work anyway, still there are several issues left. I don't recommend that.

**Comment by azizzaamir** [comment]
HookChildProcesses=0 worked. Thanks! But Why don't you recommend it? I hope everything works well.

**Comment by azizzaamir** [comment]
HookChildProcesses=0 worked. Thanks! Which issues are you talking about?

**Comment by snowie2000** [comment]
If you want to set HookChildProcesses to 0, you have to use the registry mode to ensure mactype is loaded for every new process (which is not recommended by me).

**Comment by azizzaamir** [comment]
That's ok, I won't use registry method. I just want my desktop and explorer to look pretty. And what about service mode? currently, I'm using tray mode.

**Comment by snowie2000** [comment]
You will definitely feel a delay of rendering when working in service or tray mode without child process hooking, cuz the helper process (mactray.exe) refreshes the process list periodically to check for new targets.

**Comment by extratype** [comment]
You can't get complete experience. Issues I stumbled on while I trying my best with Avast:

1. 32-bit (WoW64) processes always crashed. (with HookChildProcesses=1) I tried turning off some security features in Avast, but it didn't work.
2. Some system processes refused to load MacType. MacType DLL got unloaded and loaded again, repeatedly in cycles, wasting CPU.

**Comment by azizzaamir** [comment]
Yes, you are right. I think some apps are denying rendering that's why some applications are still not working.

**Comment by snowie2000** [comment]
Adobe Photoshop uses its own font engine and MS Word uses DirectWrite (which in theory should work with DirectWrite hook or MacTypePatch).
For compatibility issue with Avast, I'm sorry about it but that's the nature of the hook mechanism, there is probably no workaround for it.

**Comment by azizzaamir** [comment]
What if uninstall avast and retry everything, will it solve my problem? Since avast is interfering with mactype.

**Comment by extratype** [comment]
I think it would. FYI I use Bitdefender or Kaspersky.

**Comment by azizzaamir** [comment]
removed avast! now everything working like a charm! Thanks a lot!

**Comment by azizzaamir** [comment]
shall I keep directwrite enabled?

**Comment by azizzaamir** [comment]
mactype not rendering in chrome? any fix for that?

**Comment by rudolphos** [comment]
It won't work on Chrome, because they removed option to disable DirectWrite. MacType only properly works on Cent Browser and on Opera, not that on FF as well.

**Comment by IceMan81** [comment]
@rudolphos how did you get Mactype to work in Opera? Cent Browser still has the ability to disable DirectWrite, so Mactype works. 
Opera does not have the ability to disable DirectWrite (as far as I know), so curious how you got it to work in Opera.

**Comment by IceMan81** [comment]
@Jeremiidesu  Quite surprised that this worked. Have you explicitly disabled direct2d in FF to make this work? 
@snowie2000  Any ideas on how the combination of patches that @Jeremiidesu  mentioned above is working?

**Comment by rudolphos** [comment]
@IceMan81 I don't know I didn't do anything with Opera, it just works better than on regular Chrome.

I have W10 and I have only installed latest version  MacTypeInstaller_2017_0628_0.exe with Default mode.

**Comment by Jeremiidesu** [comment]
@IceMan81 I'll have to check that, I remember disabling direct2d in ff. Will also check then if MacType renders with direct2d in FF tonight!

**Comment by azizzaamir** [comment]
@IceMan81 Firefox quantum supports Mactype rendering and for Chrome I have switched to cent browser
